python/paddle/fluid/parallel_executor.py

```diff
@@ -131,6 +132,11 @@ def __init__(self,
         main = main_program
         main = main if main else framework.default_main_program()
         scope = executor.global_scope()
+        # FIXME(Yancey1989): it's a temporary approach to determinate the distribute
+        # train program, call self.bcast_param() at the end of each mini-batch.
+        self.is_dist = True if "recv" in [
```

Comment:
Can use num_trainers to test if it's distributed training

Reply:
The ```num_trainers``` looks a bit suspicious now. Is it only for nccl? I found some pserver training script working without setting num_trainers...

Reply:
It's only a temporary fix, we can insert `bcast` after `recv` op in PrallelExecuotor.

Reply:
And i think @panyx0718 is right, pserver update does not set num_trainers.